src/main/java/com/petstore/petstoreRest/service/impl/UserServiceImpl.java

```diff
@@ -1,79 +1,79 @@
 import com.petstore.petstoreRest.service.UserService;
+import jakarta.persistence.EntityNotFoundException;
 import lombok.RequiredArgsConstructor;
-import org.springframework.http.HttpStatus;
+import org.springframework.dao.DataAccessException;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 import java.util.List;
-import java.util.Optional;
 
+/**
+ * This implementation of {@link UserService} provides methods for managing user accounts.
+ */
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
 
     private final UserMapper userMapper;
+
     private final UserRepository userRepository;
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public UserDTO findByUsername(String username) {
         return userMapper.toDTO(userRepository.findByUserName(username)
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)));
-        //TODO: change to other exception
-        //                .orElseThrow(() -> new EntityNotFoundException("User with username \"" + username + "\" not found")));
+                .orElseThrow(() -> new EntityNotFoundException("User with username \"" + username + "\" not found")));
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public UserDTO saveUser(UserDTO userDTO) {
-        Optional<User> existedUser = userRepository.findByUserName(userDTO.getUserName());
-        if (existedUser.isEmpty()) {
-            User savedUser = userRepository.save(userMapper.toEntity(userDTO));
-            return userMapper.toDTO(savedUser);
-        } else {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User with such username already exists");
-            //TODO: change exception
+    public UserDTO createUser(UserDTO userDTO) {
+        try {
+            return userMapper.toDTO(userRepository.save(userMapper.toEntity(userDTO)));
+        } catch (DataAccessException e) {
+            throw new UserAlreadyExistException("User already exist");
         }
     }
 
-
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public void updateUser(String username, UserDTO userDTO) {
-        User currentUser = userRepository.findByUserName(username).orElseThrow(
-                () -> new ResponseStatusException(HttpStatus.NOT_FOUND));
-        //TODO: change exception
-
-        currentUser.setUserName(userDTO.getUserName());
-        currentUser.setFirstName(userDTO.getFirstName());
-        currentUser.setLastName(userDTO.getLastName());
-        currentUser.setEmail(userDTO.getEmail());
-        currentUser.setPassword(userDTO.getPassword());
-        currentUser.setPhone(userDTO.getPhone());
-        currentUser.setUserStatus(userDTO.getUserStatus());
+        var currentUser = userRepository.findByUserName(username)
+                .orElseThrow(() -> new EntityNotFoundException("User with username \"" + username + "\" not found"));
```

Comment:
please use String.format() static method when you construct exception message

Reply:
Changed